pkgs/build-support/dotnet/build-dotnet-module/default.nix

```
@@ -123,6 +123,8 @@ in stdenvNoCC.mkDerivation (args // {
       trap "rm -rf $src $HOME" EXIT
       pushd "$src"
 
+      rm -r global.json || true
```

I'm not really sure of this line. It's pretty hacky.

omnisharp-roslyn contains a postFixup hook which edits the global.json file - the default one upstream doesn't work with our dotnet version (it's pretty strict, I'm not sure why).

Since the `fetch-deps` passthru script uses the `src` attribute without patches, it doesnt work without removing the global.json file.

A better solution would be to use a patched source in this script, one that has all the patches and `preFixup`/`postFixup` hooks applied. I'm not sure how to do this hovewer.

---

```suggestion
      rm -f global.json
```

also please add a short comment

---

I used `srcOnly` instead, this seems like a more generic fix.